sklearn/utils/estimator_checks.py

```diff
@@ -454,8 +455,37 @@ def check_dict_unchanged(name, Estimator):
                               'Estimator changes __dict__ during %s' % method)
 
 
+def check_fit_adds_private_attributes_only(name, Estimator):
+    rnd = np.random.RandomState(0)
+    X = 3 * rnd.uniform(size=(20, 3))
+    y = X[:, 0].astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    estimator = Estimator()
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 1
+    if hasattr(estimator, "n_clusters"):
+        estimator.n_clusters = 1
+
+    set_random_state(estimator, 1)
+    dict_before_fit = estimator.__dict__.copy()
+    estimator.fit(X, y)
+
+    dict_after_fit = estimator.__dict__
+    # leave only attributes which have been set by fit
+    substracted_dicts = {k: v for k, v in dict_after_fit.items()
+                         if k not in dict_before_fit}
+
+    for val in substracted_dicts:
+        assert_true(val.startswith('_') or val.endswith('_'),
+                    ('Estimator sets invalid attributes during the fit method'
+                     'should either start with an _ or end with a _'))
+
+
+
 def check_fit2d_predict1d(name, Estimator):
-    # check by fitting a 2d array and prediting with a 1d array
+    # check by fitting a 2d array and predicting with a 1d array
```

Comment:
not relevant to the current commit, just caught my eye. Hope it's innocent enough!